### ISSUE TITLE
Fix gRPC client/server Node termination

### ISSUE THREAD

**Issue by ipetr0v** [project-oak/oak#1002]
While working on https://github.com/project-oak/oak/pull/993 I came across the following bug:
```shell
2020-05-19 12:01:59,052 DEBUG [oak_runtime::node::grpc::client] Incoming gRPC request: GrpcRequest { method_name: "/oak.examples.abitest.OakABITestService/UnaryMethod", req_msg: [18, 4, 116, 101, 115, 116], last: true }
2020-05-19 12:01:59,052 DEBUG [oak_runtime::node::grpc::client] Connecting to https://localhost:7878/
2020-05-19 12:01:59,053 DEBUG [rustls::anchors] add_pem_file processed 1 valid and 0 invalid certs
2020-05-19 12:01:59,053 ERROR [oak_runtime::node::grpc::client] Reason (factor_second): Err(RecvError(()))
2020-05-19 12:01:59,053 DEBUG [hyper::client::connect::dns] resolving host="localhost"
2020-05-19 12:01:59,054 INFO  [oak_runtime::node::grpc::client] Exiting gRPC client pseudo-Node thread
```
Looks like the gRPC pseudo-Node that uses `futures::future::select` for its termination:
https://github.com/project-oak/oak/blob/5041af16195c3798b3108c21bbe0b7abf252766d/oak/server/rust/oak_runtime/src/node/grpc/client.rs#L130-L133
receives a `RecvError(())` error (even though the Runtime doesn't send a stop notification to it).

`futures::future::select` was added in https://github.com/project-oak/oak/pull/985 in order to close Nodes that are blocked in the `tokio` execution loop.

Deleting the `futures::future::select` from prevents this error, but the Node cannot be successfully closed.

**Comment by tiziano88** [comment]
Could it be that the other side of `notify_receiver` was closed, which notifies the receiver anyways?

**Comment by ipetr0v** [comment]
Maybe, `RecvError` [page](https://docs.rs/tokio/0.2.21/tokio/sync/oneshot/error/struct.RecvError.html) doesn't specify what could cause the error.
But `notify_receiver` is saved into the `NodeInfo` so it will be closed only if the `Runtime` is closed.

**Comment by daviddrysdale** [comment]
I think this is hitting a pre-existing bug that I introduced in 2b82c464d953324e4f4917b7214fe856915e0fa1: the `NodeStopper` is being [registered](https://github.com/project-oak/oak/blob/master/oak/server/rust/oak_runtime/src/runtime/mod.rs#L1307) against the creating Node's `NodeId`, rather than the new Node's `NodeId`. 

(We got away with it previously (before #987) because a lost `JoinHandle` didn't make that much of a difference, but once the notification sender got registered against the wrong Node, things went much more awry.)

Fix included in #1016.
